packages/api/src/user.js

```diff
@@ -561,38 +561,80 @@ const notifySlack = async (
  * Get a user's payment settings.
  *
  * @param {AuthenticatedRequest} request
- * @param {Pick<import('./env').Env, 'billing'|'customers'>} env
+ * @param {Pick<BillingEnv, 'billing'|'customers'|'subscriptions'>} env
  */
 export async function userPaymentGet (request, env) {
   const userPaymentSettings = await getPaymentSettings({
     billing: env.billing,
     customers: env.customers,
+    subscriptions: env.subscriptions,
     user: { id: request.auth.user._id }
   })
-  return new JSONResponse(userPaymentSettings)
+  if (userPaymentSettings instanceof Error) {
+    switch (userPaymentSettings.code) {
+      case (new CustomerNotFound().code):
+        return new JSONResponse({
+          message: `Unexpected error fetching payment settings: ${userPaymentSettings.code}`
+        }, { status: 500 })
+      default: // unexpected error
+        throw userPaymentSettings
+    }
+  }
+  return new JSONResponse({
+    ...userPaymentSettings,
+    subscription: userPaymentSettings.subscription
+  })
 }
 
+/**
+ * @typedef {import('./utils/billing-types.js').BillingEnv} BillingEnv
+ */
+
 /**
  * Save a user's payment settings.
  *
  * @param {AuthenticatedRequest} request
- * @param {Pick<import('./env').Env, 'billing'|'customers'>} env
+ * @param {Pick<BillingEnv, 'billing'|'customers'|'subscriptions'>} env
  */
 export async function userPaymentPut (request, env) {
   const requestBody = await request.json()
-  const paymentMethodId = requestBody?.method?.id
+  const paymentMethodId = requestBody?.paymentMethod?.id
   if (typeof paymentMethodId !== 'string') {
     throw Object.assign(new Error('Invalid payment method'), { status: 400 })
   }
-  const method = { id: paymentMethodId }
+  const subscriptionInput = requestBody?.subscription
+  if (typeof subscriptionInput !== 'object') {
+    throw Object.assign(new Error(`subscription must be an object, but got ${typeof subscriptionInput}`), { status: 400 })
```

Comment:
Hadn't seen this pattern for adding "ad-hoc" fields to an `Error` before. Pretty clever :+1: